tests/framework/extensions/workloads/pods/verify.go

```diff
@@ -0,0 +1,65 @@
+package pods
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	provv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	client2 "github.com/rancher/rancher/pkg/client/generated/project/v3"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	steveV1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	DaemonsetSteveType = "apps.daemonset"
+)
+
+// VerifyReadyDaemonsetPods tries to poll the Steve API to verify the expected number of daemonset pods are in the Ready
+// state
+func VerifyReadyDaemonsetPods(t *testing.T, client *rancher.Client, cluster *v1.SteveAPIObject) error {
+	status := &provv1.ClusterStatus{}
+	err := steveV1.ConvertToK8sType(cluster.Status, status)
+	require.NoError(t, err)
+
+	daemonsetequals := false
+
+	err = wait.Poll(500*time.Millisecond, 5*time.Minute, func() (dameonsetequals bool, err error) {
+		daemonsets, err := client.Steve.SteveType(DaemonsetSteveType).ByID(status.ClusterName)
+		if err != nil {
+			return false, nil
+		}
+
+		daemonsetsStatusType := &client2.DaemonSetStatus{}
+		err = v1.ConvertToK8sType(daemonsets.Status, daemonsetsStatusType)
+		if err != nil {
+			return false, nil
+		}
+		if daemonsetsStatusType.DesiredNumberScheduled == daemonsetsStatusType.NumberAvailable {
+			return true, nil
+		}
+		return false, err
```

Comment:
I think this error going to be nil all the time because of the previous error checks. If an error occurs in the previous steps the poll gonna be triggered. So this can return err as nil too.